pkg/controller/istiocsr/fakes/fake_ctrl_client.go

```diff
@@ -0,0 +1,746 @@
+// Code generated by counterfeiter. DO NOT EDIT.
+package fakes
```

Comment:
Can we use the fake client (https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake) and cache (https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.4/pkg/cache/informertest) provided by controller-runtime rather than maintaining a separate fake client? Is there a reason not to use it in the first place?

Reply:
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake#pkg-overview has its own limitations, one such limitation which made use different library was unable to inject the required errors.